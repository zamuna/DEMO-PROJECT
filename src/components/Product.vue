<template>
  <v-card flat>
    <div v-if="isHovered">
      <b-carousel
        :id="product.id"
        v-model="slide"
        controls
        indicators
        v-if="product.images.length > 0"
        @sliding-start="onSlideStart"
        @sliding-end="onSlideEnd"
        class="d-block img-fluid"
        :img-height="product.images[0].height"
        :img-width="product.images[0].width"
        v-b-hover="handleHover"
        label-indicators="Select Previous or Next to change "
        label-next="Next Image"
        label-prev="Previous Image"
      >
        <div v-for="(image, index) in product.images" :key="index">
          <b-carousel-slide
            v-bind:img-src="image.href"
            :img-height="image.height"
            :img-width="image.width"
            :img-alt="product.name"
            :aria-labelledby="product.id"
            :aria-label="product.name"
            aria-describedby="Images for the product"
          >
          </b-carousel-slide>
        </div>
      </b-carousel>

      <img
        :id="product.id"
        :src="product.hero.href"
        :alt="product.name"
        v-else
        v-b-hover="handleHover"
        v-bind="mainProps"
        class="d-block img-fluid"
        :aria-label="product.name"
        aria-describedby="Product Image"
      />
    </div>

    <div v-else>
      <div v-if="product.hero">
        <img
          :id="product.id"
          :src="product.hero.href"
          :alt="product.name"
          v-b-hover="handleHover"
          v-bind="mainProps"
          class="d-block img-fluid"
          :aria-label="product.name"
          aria-describedby="Product Image"
          :data-isHovered="isHovered"
        />
      </div>
      <div v-else>
        <img
          :id="product.id"
          :src="product.images[0].href"
          :alt="product.name"
          v-b-hover="handleHover"
          class="d-block img-fluid"
          v-bind="mainProps"
          :aria-label="product.name"
          aria-describedby="Product Image"
        />
      </div>
    </div>

    <div class="top-left pt-2">
      <p class="text-secondary overlay" aria-label="Name of the product">
        {{ product.name }}
      </p>
    </div>
    <div class="bottom-left">
      <p
        class="text-white mt-0 mb-0 text-caption"
        aria-label="Price of the product"
      >
        {{
          product.price
            ? product.price.selling
              ? "$ " + product.price.selling
              : "$ " + product.price.regular
            : ""
        }}
      </p>
    </div>
  </v-card>

  <!-- start here -->
</template>

<script>
export default {
  data() {
    return {
      isHovered: false,
      slide: 0,
      sliding: null,
      mainProps: { blank: true, width: 363, height: 363, class: "m1" },
    };
  },
  props: {
    product: Object,
  },
  methods: {
    handleHover(hovered) {
      console.log(hovered);
      this.isHovered = hovered;
    },
    onSlideStart() {
      this.sliding = true;
    },
    onSlideEnd() {
      this.sliding = false;
    },
  },
};
</script>
