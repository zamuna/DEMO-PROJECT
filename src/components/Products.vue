<template>
  <div class="fluid-homepage">
    <v-row>
      <v-col
        v-for="(product, index) in products"
        :key="index"
        class="d-flex"
        cols="12"
        md="4"
        sm="4"
      >
        <Product :product="product" />
      </v-col>
    </v-row>
  </div>
</template>

<script>
import Product from "./Product.vue";
import axios from "axios";
export default {
  components: {
    Product,
  },
  data() {
    return {
      show: false,
      products: [],
    };
  },
  mounted: function () {
    axios
      .get(
        "http://localhost:8080/services/catalog/v4/category/shop/new/all-new/index.json"
      )
      .then((response) => {
        return response.data;
      })
      .then((jsonData) => {
        this.products = jsonData.groups;
      });
  },
};
</script>