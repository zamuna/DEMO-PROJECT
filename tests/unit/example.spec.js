import Vue from 'vue'
import Products from "@/components/Products.vue";
import Product from "@/components/Product.vue";
import { VCard, BCarousel, VBHover, BCarouselSlide } from "bootstrap-vue";
import { mount } from '@vue/test-utils'
import Vuetify from 'vuetify'


Vue.use(Vuetify)
Vue.component('b-carousel', BCarousel)
Vue.component('b-carousel-slide', BCarouselSlide)

describe('Products vue', () => {
    let wrapper = null;
    beforeEach(() => {
        wrapper = mount(Products, {
            propsData: {
                products: [{
                    id: "123",
                    name: "name1",
                    hero: "hero href",
                    images: {}
                }
                ]
            }
        });
    });

    afterEach(() => {
        wrapper.destroy();
    });

    it('renders Products.vue', () => {
        expect(wrapper.exists()).toBe(true);
    });

    it('match snapshots', () => {
        expect(wrapper).toMatchSnapshot();
    })


});

describe('Product vue', () => {
    let wrapper1 = null;
    beforeEach(() => {
        wrapper1 = mount(Product, {
            propsData: { product: { id: 'id1', name: 'name1', hero: { href: 'hero.jpg' }, images: [{ href: 'image123.jpg', name: 'img1' }, { href: 'image1.jpg', name: 'img2' }] } },
            isHovered: true,
            directives: {
                "v-card": VCard,
                "<b-carousel>": BCarousel,
                "<b-carousel-slide>": BCarouselSlide,
                'b-hover': VBHover
            }
        }

        );
    });
    afterEach(() => {
        wrapper1.destroy();
    });

    it('renders Product.vue', () => {
        expect(wrapper1.exists()).toBe(true);
    });

    it('match snapshots', () => {
        expect(wrapper1).toMatchSnapshot();
    })

    it('checks isHovered is false', () => {
        expect(wrapper1.vm.isHovered).toBe(false);
    });
    it('checks isHovered is true', () => {
        wrapper1.vm.isHovered = true;
        expect(wrapper1.vm.isHovered).toBe(true);
    });

    it('checks image is rendered', () => {
        const image = wrapper1.findAll('.v-card img');
        expect(image.exists()).toBeTruthy
    });
    it('checks slide is rendered', async () => {
        await wrapper1.setData({
            isHovered: true
        });
        const slide = wrapper1.findAll('.v-card .carousel .carousel-item img');
        expect(slide.exists()).toBeTruthy();
    });
    it('checks slide length is 2', async () => {
        await wrapper1.setData({
            isHovered: true
        });
        const slide = wrapper1.findAll('.v-card .carousel .carousel-item img');
        expect(slide.length).toBe(2)
    });

    it('image is visible', async () => {
        const image = wrapper1.find(".v-card img");
        expect(image.isVisible()).toBe(true);
    });
    it('carousel is visible', async () => {
        await wrapper1.setData({
            isHovered: true
        });
        const slide = wrapper1.findAll('.v-card .carousel .carousel-item img');
        expect(slide.isVisible()).toBe(true);

    });
});
