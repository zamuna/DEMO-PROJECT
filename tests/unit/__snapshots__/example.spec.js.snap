// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product vue match snapshots 1`] = `
<div class="v-card v-card--flat v-sheet theme--light">
  <div>
    <div><img id="id1" src="hero.jpg" alt="name1" aria-label="name1" aria-describedby="Product Image" blank="true" width="363" height="363" class="d-block img-fluid m1"></div>
  </div>
  <div class="top-left pt-2">
    <p aria-label="Name of the product" class="text-secondary overlay">
      name1
    </p>
  </div>
  <div class="bottom-left">
    <p aria-label="Price of the product" class="text-white mt-0 mb-0 text-caption">

    </p>
  </div>
</div>
`;

exports[`Products vue match snapshots 1`] = `
<div class="fluid-homepage" products="[object Object]">
  <div class="row"></div>
</div>
`;
